README.md
# 2021-project-02-help-wedding
# Help-Wedding

Help wedding merupakan sebuah aplikasi yang bertujuan untuk menyediakan keperluan
untuk pesta pernikahan. Dimana aplikasi ini akan membantu menentukan kebutuhan
penyelenggaraan acara pernikahan dengan memperoleh berbagai informasi sekaligus melakukan
pemesanan gedung, tata rias ,makanan,dan dekorasi. Help wedding akan memberikan kebebasan
untuk klien agar dapat memilih beberapa opsi yang disediakan sesuai dengan yang diinginkan oleh
klien agar klien dapat menyesuaikan dengan budget yang dimilikinya.
Aplikasi ini akan menyediakan 4 kategori yaitu :
1. Layanan gedung menyediakan 3 opsi , diantara lain:
- 1000 orang dengan area parkir
- 800 orang dengan area parkir
- 500 orang tanpa area parkir
2. Layanan tata rias disediakan 4 opsi , diantara lain:
- Untuk pengantin dan keluarga, bunga melati, dan bouquet bunga hidup
- Untuk pengantin dan keluarga, bunga melati dan bouquet bunga plastik
- Untuk pengantin, bunga melati full dan bouquet bunga hidup
- Untuk pengantin, bunga melati dan bouquet bunga plastik
3. Layanan penyediaan makanan tersedia 3 opsi , diantara lain:
- 800 tradisional + 200 nasional
- 600 tradisional + 200 nasional
- 300 tradisional + 200 nasional
4. Layanan dekorasi pelaminan menyediakan 2 opsi , diantara lain:
- Pelaminan dengan Full Bunga Hidup
- Pelaminan dengan Full Bunga Plastik

Untuk dapat berinteraksi dengan sistem, Admin harus melakukan login terlebih dahulu dengan menginput 

- username: admin
- password: login123
 
Setelah melakukan login , admin dapat memilih menu yang ada pada sistem seperti :

- Lihat Daftar Layanan

Menu ini akan menampilkan daftar layanan berupa idOpsi,namaOpsi, nama layanan, harga opsi, dan stok layanan.
- Edit Data Layanan

Pada menu ini admin dapat melakukan edit data layanan seperti memperbaharui stok layanan apabila ada yang melakukan pemesanan.

- Lihat Daftar Pemesanan

Menu ini akan menampilkan daftar client yang melakukan pemesanan. Atribut yang ditampilkan berupa idPesanan, nama klien, idOpsi, tanggal pesanan,tanggal
pesta , dan alamat.
- Input Data Pemesanan

Pada menu ini admin menginput data apabila ada client yang ingin melakukan pemesanan. Apabila client ingin melakukan pemesanan, maka admin akan terlebih dahulu menginput tanggal pesanan setelah itu menginput tanggal pesta. Apabila tanggal pesta ke tanggal pesanan rentang nya kurang dari 2 bulan maka client tidak dapat melakukan pemesanan , sebaliknya apabila rentang waktunya lebih dari 2 bulan maka client dapat melakukan pemesanan. Atribut yang diminta sistem pada menu ini berupa idPesanan,nama klien,idOpsi, tanggal pesanan,tanggal pesta dan alamat.

- Edit Data Pemesanan

Pada menu ini, admin dapat mengganti informasi berupa atribut yang ada pada data pemesanan dari client dengan menggunakan idPesanan sebagai pengenal masing
masing informasi dari client.
- Menghapus Data Pemesanan

Pada menu ini ,apabila admin salah menginput data pesanan maka admin dapat memilih menu ini dan melakukan penghapusan data pesanan client dengan menggunakan idPesanan sebagai pengenal masing masing informasi dari client.

- Form Pembayaran

Pada menu ini, setelah melakukan pemesanan, sistem akan menampilkan total biaya pemesanan dan akan menghitung 25% dari total harga sebagai uang muka
yang harus dibayar langsung. Pembayaran kedua sebesar 25% dibayarkan 1 bulan setelah pembayaran uang muka pertama, pelunasan paling lambat 14 hari sebelum pesta. Admin akan menginput nama client kemudian sistem akan menampilkan masing-masing idopsi, harga pembayaran, jumlah uang muka , jumlah pembayaran kedua, dan jumlah pelunasan yang harus dibayar. Form pembayaran ini akan diberikan kepada client dalam bentuk nota . Dimana isi dari nota berupa nama client, id pesanan dan subtotal.

- Melakukan Pembayaran

Pada menu ini, apabila client ingin membayar pesanan maka admin akan memasukkan id pesanan, nama client, subtotal, tanggal pembayaran, dan jenis pembayaran. Dimana jenis pembayaran ada 3 , pembayaran uang muka, pembayaran kedua setelah uang muka dibayar, dan terakhir pelunasan.

- Validasi pembayaran

Pada menu ini, setelah client selesai membayar, admin akan mengisi form seperti struk pembayaran dan struk pembayaran ini akan diberikan kepada client sebagai
bukti bahwa client telah berhasil melakukan pembayaran.

- Riwayat pembayaran

Pada menu ini, sistem akan menampilkan riwayat daftar dari semua pembayaran yang berupa id pesanan , nama client, subtotal, jenis pembayaran, dan tanggal
pembayaran.


changelog.txt
menambahkan class Pembayaran
menambahkan class Layanan
menambahkan code pada class layanan
menambahkan code pada class pembayaran
menambahkan code pada class layanan, inventory
menambahkan method pada class Admin
menambahkan code pada class InventoryHelpWedding dan Pemesanan
mengganti nama kelas datalayanan menjadi layanan
memperbaiki code pada kelas Pembayaran
memperbaiki code pada kelas Pembayaran 2
memperbaiki code pada class Admin
memperbaiki code pada kelas Pemesanan
memperbaiki code pada kelas Pemesanan 2
memperbaiki code pada kelas pembayaran
memperbaiki code pada class datalayanan
memperbaiki code pada kelas pemesanan 
memperbaiki code pada class datalayanan
memperbaiki code pada class Admin
memperbaiki code pada kelas pembayaran 4
memperbaiki code pada kelas pemesanan
memperbaiki error pada changelog
menambahkan code pada kelas Pemesanan 3 dan InventoryHelpWedding
memperbaiki method pada class Admin
memperbaiki method pada kelas pembayaran dan inventori
memperbaiki tampilan output riwayat pembayaran
memperbaiki code pada kelas Pemesanan 
memperbaiki code pada class datalayanan
memperbaiki code pada kelas Pemesanan 
memperbaiki code pada class Admin
menambahkan code pada class InventoryHelpWedding
memperbaiki code pada class InventoryHelpWedding
memperbaiki code pada class pembayaran
memperbaiki code pada class pembayaran
memperbaiki code pada class datalayanan
memperbaiki code pada class pembayaran
memperbaiki code pada class InventoryHelpWedding
memperbaiki code pada class pembayaran

src/helpwedding/Pemesanan.java
package helpwedding;

import java.util.*;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.PreparedStatement;

public class Pemesanan {
    private String idPesanan;
    private String namaClient;
    private String idOpsi;
    private String tanggalPesanan;
    private String tanggalPesta;
    private String alamat;
   
    
    //Melakukan koneksi ke database
    public Connection connect() {
        Connection connect = null;
        try {
            connect = DriverManager.getConnection("jdbc:sqlite:F:/Semester 4/PBO/Praktikum/2021-project-02-help-wedding/Layanan.db");
        } catch (Exception e) {
            System.out.println(e);
        }
        return connect;
    }
    
  //fungsi untuk melihat data dalam database
    public void melihatDaftarpesanan() {
        String query = "Select * from pemesanan";
        try {
            Connection connect = this.connect(); //Memanggil method koneksi yang sebelumnya didefenisikan
            PreparedStatement stmt = connect.prepareStatement(query);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                System.out.println(rs.getString("idPesanan") + "\t\t"
                        + rs.getString("namaClient") + "\t\t\t"
                        + rs.getString("idOpsi") + "\t\t"
                        + rs.getString("tanggalPesanan") + "\t\t"
                        + rs.getString("tanggalPesta") + "\t\t"
                        + rs.getString("alamat"));
            }
        } catch (Exception e) {
            System.out.println(e);
        }
    }

    
  //fungsi untuk menambahkan data ke dalam database
    public void melakukanPemesanan(String idPesanan, String namaClient, String idOpsi, String tanggalPesanan, String tanggalPesta, String alamat) {
        String query = "insert into pemesanan values(?,?,?,?,?,?)";
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, idPesanan);
            stmt.setString(2, namaClient);
            stmt.setString(3, idOpsi);
            stmt.setString(4, tanggalPesanan);
            stmt.setString(5, tanggalPesta);
            stmt.setString(6, alamat);
            stmt.executeUpdate();
        } catch (Exception e) {
            System.out.println(e);
        }
    }     
    
   //fungsi untuk mengubah data pemesanan
     public void mengubahPemesanan(String idPesananLama, String idPesananBaru, String namaClient, String idOpsi, String tanggalPesanan, String tanggalPesta, String alamat) {
         String query = "update pemesanan set idPesanan = ?, namaClient = ?, idOpsi =?, tanggalPesanan =?, tanggalPesta =?, alamat =?  where idPesanan = ?";
         try {
             Connection connect = this.connect();
             PreparedStatement stmt = connect.prepareStatement(query);
             stmt.setString(1, idPesananBaru);
             stmt.setString(2, namaClient);
             stmt.setString(3, idOpsi);
             stmt.setString(4, tanggalPesanan);
             stmt.setString(5, tanggalPesta);
             stmt.setString(6, alamat);
             stmt.setString(7, idPesananLama);
             stmt.executeUpdate();
         } catch (Exception e) {
             System.out.println(e);
         }
     }
    
   //fungsi untuk menghapus salah satu data pemesanan berdasarkan idPesanan
     public void menghapusPemesanan(String idPesanan) {
         String query = "delete from pemesanan where idPesanan =?";
         try {
             Connection connect = this.connect();
             PreparedStatement stmt = connect.prepareStatement(query);
             stmt.setString(1, idPesanan);
             stmt.executeUpdate();
         } catch (Exception e) {
             System.out.println(e);
         }
     }
}


src/helpwedding/DataLayanan.java
package helpwedding;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.PreparedStatement;

public class DataLayanan {

    private String namaLayanan;
    private String idOpsi; 
    private String namaOpsi;
    private Double hargaOpsi;
    private Integer stokLayanan;

  
    // Constructor with a parameter
    public DataLayanan (){
        this.namaLayanan = namaLayanan;
        this.idOpsi = idOpsi;
        this.namaOpsi = namaOpsi;
        this.hargaOpsi = hargaOpsi;
        this.stokLayanan = stokLayanan;

    }

    //Melakukan koneksi ke database
    public Connection connect() {
        Connection connect = null;
        try {
            connect = DriverManager.getConnection("jdbc:sqlite:F:/Semester 4/PBO/Praktikum/2021-project-02-help-wedding/Layanan.db");
        } catch (Exception e) {
            System.out.println(e);
        }
        return connect;
    }
     

    //fungsi untuk melihat semua DaftarLayanan ke dalam database
    public void getAllLayanan(){ 
        String query = "Select * From Layanan";
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                System.out.println( rs.getString("idOpsi") + "\t\t"
                        + rs.getString("namaLayanan") + "  " + "\t"
                        + rs.getString("namaOpsi") + "   " + " " + "\t" 
                        + rs.getInt("hargaOpsi") + "  "+ "\t"
                        + rs.getInt("stokLayanan"));
            }
        } catch (Exception e) {
            System.out.println(e);
        }
        }
    
     //fungsi untuk memperbaharui data Layanan setelah dilakukan pemesanan

    public void updateLayanan(String idOpsiLama, String idOpsiBaru, String namaLayanan, String namaOpsi, double hargaOpsi, int stokLayanan) {
        String query = "update Layanan set idOpsi = ?, namaLayanan = ?, namaOpsi =?, hargaOpsi =?, stokLayanan = ?  where idOpsi = ?";
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, idOpsiBaru);
            stmt.setString(2, namaLayanan);
            stmt.setString(3, namaOpsi);
            stmt.setDouble(4, hargaOpsi);
            stmt.setInt(5, stokLayanan);
            stmt.setString(6, idOpsiLama);
            stmt.executeUpdate();
        } catch (Exception e) {
            System.out.println(e);
        }
    }




}

src/helpwedding/InventoryHelpWedding.java
package helpwedding;

import java.util.*;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.Scanner;
import java.sql.SQLException;
import java.time.LocalDate;
import java.time.Month;
import java.time.Period;
import java.util.Scanner;

public class InventoryHelpWedding{
	public static void main(String[] args) {
		
		System.out.println("=====================================================");
		System.out.println("Selamat Datang Di Help Wedding");
		DateFormat formatTanggal = new SimpleDateFormat(" E, dd-MMM-yyyy HH:mm:ss");
		Date date = new Date();
		System.out.println(formatTanggal.format(date));
		System.out.println("=====================================================");
        System.out.println("\n");
        
        Scanner input = new Scanner(System.in);
        Admin admin = new Admin();
        
        Main:while(true){
            System.out.println("==++ Selamat Datang Di HelpWedding ++==");
            System.out.println("1.Masuk Sebagai Admin");
            System.out.println("0.Keluar");
            System.out.print("Masukkan pilihan : ");
            int pilihan = input.nextInt();
            
            if(pilihan == 1){
                System.out.println("Silahkan masukkan username dan password");
                System.out.print("Masukkan username : ");              
                String username = input.next();      
                
                System.out.print("Masukkan password : ");
                String password = input.next();
                if(admin.login(username, password) == 1){
                    while(true){

                        pilihan = 0;
                        System.out.println("\n\n |||=== Selamat Datang Admin HelpWedding  ===|||");
                        System.out.println("1.Lihat Daftar Layanan");
                        System.out.println("2.Edit Data Layanan");
                        System.out.println("3.Lihat Daftar Pemesanan");
                        System.out.println("4.Input Data Pemesanan");
                        System.out.println("5.Edit Data Pemesanan");
                        System.out.println("6.Hapus Data Pemesanan");
                        System.out.println("7.Form Pembayaran");
                        System.out.println("8.Melakukan Pembayaran");
                        System.out.println("9.Validasi Pembayaran");
                        System.out.println("10.Riwayat Pembayaran");
                        System.out.println("11.Kembali ke menu utama");
                        
                        System.out.print("Masukkan pilihan : ");
                        pilihan = input.nextInt();
                        
                        if(pilihan == 1){
                            System.out.println("==================================================== Daftar Layanan ===============================================");
                            System.out.println("id Opsi \tNama Layanan \t\tNama Opsi \t\t\t\t\tHarga Opsi \tStok Layanan");
                        	System.out.println("=====================================================================================================================");
                            (new DataLayanan()).getAllLayanan();
                        }
                        
                        else if(pilihan == 2){
                        	(new DataLayanan()).updateLayanan("G001","G001","Gedung","Kapasitas 1000 orang yang dilengkapi  area  parkir\t",50000000,4);
                        }
                        
                        else if(pilihan == 3){
                        	System.out.println("==================================================== Daftar Pemesanan ===============================================");
                        	System.out.println("id Pesanan \tNama Client \t\tid Opsi \tTanggal Pesanan \tTanggal Pesta\t\tAlamat");
                        	System.out.println("=====================================================================================================================");
                        	(new Pemesanan()).melihatDaftarpesanan();
                        }
                        
                        else if(pilihan == 4){
                        	System.out.println("Menu Input pemesanan");

                        	Scanner sc = new Scanner(System.in);

                        	System.out.print("Masukkan Tanggal Pemesanan(Format yyyy-MM-dd): ");
                        	String tanggalPesanan = sc.next();

                        	System.out.print("Masukkan Tanggal Pesta(Format yyyy-MM-dd): ");
                        	String tanggalPesta = sc.next();

                        	LocalDate awal = LocalDate.parse(tanggalPesanan);
                        	LocalDate akhir = LocalDate.parse(tanggalPesta);


                        	System.out.println(awal);
                        	System.out.println(akhir);

                        	Period jumlahHari = Period.between(awal, akhir);

                        	System.out.printf("\nRentang waktu = %d tahun, %d bulan dan %d hari \n\n", 
                        	jumlahHari.getYears(), jumlahHari.getMonths(), jumlahHari.getDays());
                        	    
                        	    if( jumlahHari.getMonths() > 1 ) {
                        	        System.out.println("Pemesanan dapat dilanjutkan");
                        	        
                        	        System.out.print("Masukkan ID Pesanan: ");
                        	        String idPesanan = sc.next();

                        	        System.out.print("Masukkan Nama Client: ");
                        	        String namaClient = sc.next();
                        	        
                        	          System.out.print("Masukkan ID Opsi: ");
                        	        String idOpsi = sc.next();

                        	        System.out.print("Masukkan Alamat: ");
                        	        String alamat = sc.next();
                        	         
                        	        
                        	        (new Pemesanan()).melakukanPemesanan(idPesanan,namaClient,idOpsi,tanggalPesanan,tanggalPesta,alamat);     
                        	        
                        	    }
                        	    else if (jumlahHari.getYears() >= 1){
                        	        System.out.println("Pemesanan dapat dilanjutkan");
                        	        
                        	        System.out.print("Masukkan ID Pesanan: ");
                        	        String idPesanan = sc.next();

                        	        System.out.print("Masukkan Nama Client: ");
                        	        String namaClient = sc.next();
                        	        
                        	         System.out.print("Masukkan ID Opsi: ");
                        	        String idOpsi = sc.next();

                        	        System.out.print("Masukkan Alamat: ");
                        	        String alamat = sc.next();
                        	       

                        	        (new Pemesanan()).melakukanPemesanan(idPesanan,namaClient,idOpsi,tanggalPesanan,tanggalPesta,alamat);
                        	    }
                        	    
                        	    else {
                        	        System.out.println("Maaf, Pemesanan paling lama dilakukan 2 bulan sebelum pesta dilakukan");
                        	    }    	
                        }
                        
                        else if(pilihan == 5){
                        	  System.out.println("Menu Edit pemesanan");

                              Scanner sc = new Scanner(System.in);

                              System.out.print("Masukkan ID PesananLama: ");
                              String idPesananLama = sc.next();

                              System.out.print("Masukkan ID PesananBaru: ");
                              String idPesananBaru = sc.next();

                              System.out.print("Masukkan Nama Client: ");
                              String namaClient = sc.next();
                              
                              System.out.print("Masukkan ID Opsi: ");
                              String idOpsi = sc.next();
                              
                              System.out.print("Masukkan Tanggal Pemesanan: ");
                              String tanggalPesanan = sc.next();

                              System.out.print("Masukkan Tanggal Pesta: ");
                              String tanggalPesta = sc.next();

                              System.out.print("Masukkan Alamat: ");
                              String alamat = sc.next();
                              
                              (new Pemesanan()).mengubahPemesanan(idPesananLama,idPesananBaru,namaClient,idOpsi,tanggalPesanan,tanggalPesta,alamat);     	
                        }
                        
                        else if(pilihan == 6){
                        	System.out.println("Menu Hapus Pemesanan");
                            Scanner sc = new Scanner(System.in);

                            System.out.print("Masukkan ID Pesanan yang akan dihapus: ");
                            String idPesanan = sc.next();
                            
                            (new Pemesanan()).menghapusPemesanan(idPesanan);     	
                        }
                        
                        else if(pilihan == 7){
                        	Scanner sc = new Scanner(System.in);
                        	System.out.println("Masukkan nama client: ");
                        	String namaClient = sc.next();
                        	System.out.println("============================");
                        	System.out.println("id Opsi \tHarga Opsi ");
                        	System.out.println("============================");
                        	(new Pembayaran()).melihatHargaPesananClient(namaClient);
                    		(new Pembayaran()).totalpembayaran(namaClient);
                    		(new Pembayaran()).PembayaranUangMuka(namaClient);
                    		(new Pembayaran()).PembayaranKedua(namaClient);
                    		(new Pembayaran()).Pelunasan(namaClient);
                        	       	
                        }
                        
                        else if(pilihan == 8){
                        	Scanner sc = new Scanner(System.in);
                        	System.out.print("Masukkan ID Pesanan: ");
                	        String idPesanan = sc.next();

                	        System.out.print("Masukkan Nama Client: ");
                	        String namaClient = sc.next();
                	        
                	          System.out.print("Masukkan Subtotal: ");
                	        int subtotal = sc.nextInt();

                	        System.out.print("Masukkan Jenis Pembayaran: ");
                	        String jenis_pembayaran = sc.next();
                	        
                	        System.out.print("Masukkan tanggal_pembayaran: ");
                	        String tanggal_pembayaran = sc.next();
                	        
                	        (new Pembayaran()).melakukanPembayaran(idPesanan,namaClient,subtotal,jenis_pembayaran,tanggal_pembayaran);
                        	
                        	
                        }
                        
                        else if(pilihan == 9){
                        	Pembayaran pembayaran = new Pembayaran();
                    		pembayaran.validasiPembayaran(pembayaran);
                        	
                        } 
                        
                        else if(pilihan == 10){
                        	System.out.println("==================================================== Riwayat Pembayaran ===============================================");
                        	System.out.println("id Pesanan \tNama Client \t\tSubtotal \tJenis Pembayaran \tTanggal Pembayaran");
                        	System.out.println("=======================================================================================================================");
                        	(new Pembayaran()).melihatRiwayatPembayaran();
                        }
                        
                        else if(pilihan == 11){
                        	continue Main;
                        }
            
                        
                    }
                } 
                
            }
            
            else if (pilihan == 0) {
            System.out.println("Terimakasih sudah berkunjung ");
            System.exit(0);
        }
            else {
            	System.out.println("Pilihan Anda Salah!!");
            }
	}
	
}
}

src/helpwedding/Pembayaran.java
package helpwedding;

import java.util.*;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.PreparedStatement;

public class Pembayaran {
	private String idPesanan;
    private String namaKlien;
    private Integer subtotal;
    private String jenis_pembayaran;
    private String tanggal_pembayaran;

    
    //mengkoneksikan ke database
    public Connection connect() {
       
        Connection connect = null;
        try { 
            connect = DriverManager.getConnection("jdbc:sqlite:F:/Semester 4/PBO/Praktikum/2021-project-02-help-wedding/Layanan.db");
        } catch (Exception e) {
            System.out.println(e);
        }
        return connect;
    }

    //fungsi untuk mendapatkan semua id Opsi dan harga Opsi yang dipesan oleh client dengan nama client = ...
    public void melihatHargaPesananClient(String namaClient) {
        String query = "SELECT * FROM  pemesanan  NATURAL JOIN layanan WHERE namaClient = ?";
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, namaClient);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                System.out.println(rs.getString("idOpsi") + "\t\t"
                		          +rs.getInt("hargaOpsi"));
            }
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    
    //fungsi untuk mendapatkan total harga yang harus dibayar oleh client
    public void totalpembayaran(String namaClient){
    	String query = "SELECT sum(hargaOpsi) FROM  pemesanan  NATURAL JOIN layanan WHERE namaClient = ? ";
    	int sum = 0;
        try { 
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, namaClient);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                int c = rs.getInt(1);
                sum = sum + c;
              }
              System.out.println("Total Pembayaran = " + sum);
           
            
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    
  //fungsi untuk mendapatkan jumlah uang muka sebesar 25% dari total yang harus dibayar oleh client setelah melakukan pemesanan
    public void PembayaranUangMuka(String namaClient){
        String query = "SELECT sum(hargaOpsi) FROM  pemesanan  NATURAL JOIN layanan WHERE namaClient = ?";
        int sum = 0;
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, namaClient);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
            	int c = rs.getInt(1);
                sum = sum + c;
              }
              System.out.println("Uang muka yang harus dibayar : " + (sum*25)/100);
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    
    //fungsi untuk mendapatkan jumlah uang pembayaran kedua sebesar 25% setelah 1 bulan membayar uang muka.
    public void PembayaranKedua(String namaClient){
    	String query = "SELECT sum(hargaOpsi) FROM  pemesanan  NATURAL JOIN layanan WHERE namaClient = ?";
        int sum = 0;
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, namaClient);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
            	int c = rs.getInt(1);
                sum = sum + c;
              }
              System.out.println("Cicilan yang harus dibayar setelah 1 bulan pembayaran uangmuka = " + ((sum-(sum*25)/100)*25)/100);
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    
    //fungsi untuk mendapatkan jumlah uang yang harus dilunaskan 14 hari sebelum pesta berlangsung.
    public void Pelunasan(String namaClient){
    	String query = "SELECT sum(hargaOpsi) FROM  pemesanan  NATURAL JOIN layanan WHERE namaClient = ?";
        int sum = 0;
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, namaClient);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
            	int c = rs.getInt(1);
                sum = sum + c;
              }
              System.out.println("Sisa dari uang muka dan cicilan kedua yang harus dilunaskan 14 hari sebelum pesta = " 
              + (sum - ((sum*25)/100 + ((sum-(sum*25)/100)*25)/100)));
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    

    //fungsi untuk melakukan validasi pembayaran, hasilnya akan berupa struk pembayaran yang akan diberikan kepada client.
    public void validasiPembayaran(Pembayaran date){
    
        Scanner scan = new Scanner (System.in);
        System.out.print("Masukkan nama klien  : ");
        String namaClient = scan.next();  
        System.out.print("Masukkan id Pesanan : ");
        String idPesanan = scan.next();   
        System.out.print("Masukkan subtotal  : ");
        Integer subtotal = scan.nextInt(); 
        System.out.print("Masukkan jenis pembayaran :");
        String jenis_pembayaran = scan.next();
        System.out.println("Nama klien		    : "+namaClient);  
        System.out.println("Id Pesanan		    : "+idPesanan);  
        System.out.println("Subtotal		    : "+subtotal);  
        System.out.println("Jenis pembayaran    : "+jenis_pembayaran);
        System.out.println("Tanggal Pembayaran	: "+date.getTanggal()); 
}
    
   
  //fungsi untuk menambahkan pembayaran
    public void melakukanPembayaran(String idPesanan, String namaClient, int subtotal, String jenis_pembayaran, String tanggal_pembayaran) {
        String query = "insert into pembayaran values(?,?,?,?,?)";
        try {
            Connection connect = this.connect();
            PreparedStatement stmt = connect.prepareStatement(query);
            stmt.setString(1, idPesanan);
            stmt.setString(2, namaClient);
            stmt.setInt(3, subtotal);
            stmt.setString(4, jenis_pembayaran);
            stmt.setString(5, tanggal_pembayaran);
            stmt.executeUpdate();
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    
    //fungsi untuk melihat riwayat pembayaran
    public void melihatRiwayatPembayaran() {
        String query = "Select * from pembayaran";
        try {
            Connection connect = this.connect(); //Memanggil method koneksi yang sebelumnya didefenisikan
            PreparedStatement stmt = connect.prepareStatement(query);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                System.out.println(rs.getString("idPesanan") + "\t\t"
                        + rs.getString("namaClient") + "\t\t\t"
                        + rs.getInt("subtotal") + "\t\t"
                        + rs.getString("jenis_pembayaran") + "\t\t"
                        + rs.getString("tanggal_pembayaran"));
            }
        } catch (Exception e) {
            System.out.println(e);
        }
    }
    
    String getTanggal(){
        DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd");  
          Date date = new Date();  
          return dateFormat.format(date);
     }   
      
    }